### PR TITLE
rustdoc-search: use smart binary search in bitmaps

### PR DESCRIPTION
**Pull request by notriddle**
Addresses a comment from [jsha's benchmarking], where the `contains` function showed up in the profiler. This commit pulls it from about 5% of the runtime to about 0.5%.

Before: https://share.firefox.dev/3ANVjon

After: https://share.firefox.dev/3OeM3gk

[jsha's benchmarking]: https://rust-lang.zulipchat.com/#narrow/channel/266220-t-rustdoc/topic/search.20profiling/near/481868761

### CODE REVIEW

File: src/librustdoc/html/static/js/search.js [hunk 1]
```diff
@@ -1053,9 +1053,17 @@ class RoaringBitmap {
     contains(keyvalue) {
         const key = keyvalue >> 16;
         const value = keyvalue & 0xFFFF;
-        for (let i = 0; i < this.keys.length; ++i) {
-            if (this.keys[i] === key) {
-                return this.containers[i].contains(value);
+        let left = 0;
```
Comment (GuillaumeGomez) [inline src/librustdoc/html/static/js/search.js:1056]: Please add some code comments to explain the algorithm used and in particular why we don't simply iterate over the items.

Comment (notriddle) [reply src/librustdoc/html/static/js/search.js:1069]: Okay, I've added some comments.

Comment (GuillaumeGomez) [reply src/librustdoc/html/static/js/search.js:1069]: Much appreciated, thanks!

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks! r=me once CI pass.

**Pull request merged**